### PR TITLE
fix: handling of double and integer

### PR DESCRIPTION
**Pull request by apulbere**
## This PR

- stores Number instead of Double
- removes custom conversion from double to int and relies instead on API that respects Java Specs https://docs.oracle.com/javase/specs/jls/se7/html/jls-5.html#jls-5.1.3

### Notes

This is a proposal

### CODE REVIEW

**Comment by beeme1mr** [comment]
Hey @apulbere, thanks for the PR. Someone will review it shortly.

File: src/test/java/dev/openfeature/sdk/ValueTest.java [hunk 1]
```diff
@@ -66,11 +66,11 @@ class Something {}
     }
 
     @Test public void numericArgShouldReturnDoubleOrInt() {
-        double innerDoubleValue = .75;
+        double innerDoubleValue = 1.75;
         Value doubleValue = new Value(innerDoubleValue);
         assertTrue(doubleValue.isNumber());
-        assertEquals(1, doubleValue.asInteger());     // should be rounded
-        assertEquals(.75, doubleValue.asDouble());
+        assertEquals(1, doubleValue.asInteger());     // the double value represented by this object converted to type int
```
Comment (toddbaert) [inline src/test/java/dev/openfeature/sdk/ValueTest.java:72]: Thanks for linking the spec related to this rounding "toward zero". I was not aware of this nuance.

**Review by toddbaert — APPROVED** [review]

**Review by Kavindu-Dodan — APPROVED** [review]
Nice work. Yes it makes sense to store Number type instead of double.

**Comment by beeme1mr** [comment]
@justinabrahms any concerns?

**Pull request merged**